subsys/bluetooth/mesh/sensor_cli.c

```diff
@@ -573,13 +573,17 @@ int bt_mesh_sensor_cli_desc_all_get(struct bt_mesh_sensor_cli *cli,
 
 	struct list_rsp list_rsp = {
 		.sensors = sensors,
-		.count = *count,
+		.count = count ? *count : 0,
 	};
 
 	err = model_ackd_send(cli->model, ctx, &msg, sensors ? &cli->ack_ctx : NULL,
 			      BT_MESH_SENSOR_OP_DESCRIPTOR_STATUS, &list_rsp);
+	if (!count || err) {
```

Comment:
What about `if (count && !err)` around the assignment instead, to avoid multiple returns here? Makes more sense to me at least. Also, might use explicit count != NULL here.